torch/csrc/utils/functional.h

```diff
@@ -18,41 +14,26 @@ namespace torch {
 // No explicit template parameters are required.
 
 // Overload for explicit function and ArrayRef
-template<typename F, typename T, typename R = typename std::result_of<F(T)>::type>
-inline std::vector<R> fmap(at::ArrayRef<T> inputs, const F& fn) {
-  std::vector<R> r;
+template<typename F, typename T>
+inline auto fmap(const T& inputs, const F& fn) -> std::vector<decltype(fn(*inputs.begin()))> {
+  std::vector<decltype(fn(*inputs.begin()))> r;
```

Comment:
If you still have this change paged in your head, I am curious to know how this solves the ambiguous conversion problem that the code previously had :)